enhancements/olm/catalogd-console-api.md

```diff
@@ -0,0 +1,670 @@
+but the endpoint provides complete catalog content in a single transaction.  In performance testing with 
+proof of concept OCP console work, it proves insufficiently responsive and requires user-agent-side results
+ caching of large amounts of catalog data.   
+We would like to introduce a new HTTPS endpoint to provide more fine-grained and lower-latency access to 
+catalog contents. 
+
+<!--
+The `Summary` section is important for producing high quality
+user-focused documentation such as release notes or a development roadmap. It
+should be possible to collect this information before implementation begins in
+order to avoid requiring implementors to split their attention between writing
+release notes and implementing the feature itself.
+
+Your summary should be one paragraph long. More detail
+should go into the following sections.
+--> 
+
+## Motivation
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+-->
+
+The existing `/api/v1/all` endpoint returns the entire FBC, which console or other clients then need 
+to process in order to retrieve relevant information. This is very inefficient for 
+accesses which can be satisfied by a much smaller record set. 
+
+<!--
+redhat-operator-index:    55MB
+certified-operator-ind:   32MB
+community-operator-index: 29MB
+operatorhub.io:           21MB          <--- the benchmark 4-10s duration
+redhat-marketplace-index:  9MB
+-->
+
+Requiring OCP console to make the full catalog available to users incurs a 4-10 second delay _per catalog_ even 
+under optimal network conditions, plus large local cache coherency challenges to avoid additional retrievals to 
+fulfill further use-cases as a user makes progress towards selecting content to install.
+
+
+### User Stories
+
+* As a console user, I would like to be able to discover available packages for installation.
+* As a console user, I would like to be able to request only catalog context relevant to a specific package.
+* As a console user, I would like to be able to discover available package updates.
+* As a console user, I would like to be able to view details for a specific channel for a package.
+* As a console user, I would like to be able to discover all bundle versions associated with a package 
+(and optionally a specific channel).
+* As a console user, I would like to be able to view details for a specific bundle version.
+
+
+### Goals
+
+* Provide an HTTPS catalog content discovery endpoint which fulfills targeted catalog queries with a minimum record set.
+
+<!--
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+-->
+
+### Non-Goals
+
+* Redesigning FBC schema to facilitate additional efficiencies.
```

Comment:
I am very curious what @spadgett and @TheRealJon think about whether this enhancement, on its own (i.e. without further FBC schema evolution), will be a significant enough improvement for Console's use cases, that it is reasonable to keep schema evolution out-of-scope.

I ask because we may need to include the schema evoluation in scope in order to reference it in a graduation criteria for taking the combined OLM and Console changes for OLMv1 support to GA.

Reply:
It's hard to know for sure whether this will make significant improvements to the network latency issues. We still require almost all of the FBC data in order to populate the catalog view as designed. We hope that making more strategic asynchronous requests will help. The data model issues still remain.